base/ext/scope_guard_safe.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <ext/scope_guard.h>
+#include <common/logger_useful.h>
+#include <Common/MemoryTracker.h>
+
+/// Same as SCOPE_EXIT() but try/catch/tryLogCurrentException any exceptions.
+#define SCOPE_EXIT_SAFE(...) SCOPE_EXIT(             \
+    try                                              \
+    {                                                \
+        __VA_ARGS__;                                 \
+    }                                                \
+    catch (...)                                      \
+    {                                                \
+        tryLogCurrentException(__PRETTY_FUNCTION__); \
+    }                                                \
+)
+
+/// Same as SCOPE_EXIT() but:
+/// - block the MEMORY_LIMIT_EXCEEDED errors,
+/// - try/catch/tryLogCurrentException any exceptions.
+#define SCOPE_EXIT_MEMORY_SAFE(...) SCOPE_EXIT(                   \
```

Comment:
How should we choose these variants? Perhaps we can put some when-to-use example in the comment.

Reply:
Hm, I though that describing what they do is enough, OK, will try to give a sensible comment.

Reply:
Done, I hope that it will bring some clarification.